pipeline/processor/postprocessor.py

```diff
@@ -0,0 +1,663 @@
+            filename: The filename of the document.
+
+        Returns:
+            The processed document.
+        """
+        pass
+
+
+class HyphenationPostProcessor(PostProcessor):
+    """
+    Post processor to join words that are separated into separate blocks due
+    to hyphenation at line breaks.
+    """
+
+    @staticmethod
+    def _rewrap_hyphenated_words(li: list) -> list:
+        """
+        Reorganise a list of strings so that word fragments separated
+        by hyphens or em dashes to start new lines are joined into a single
+        word if the word fragments have no meaning by themselves.
+
+        Args:
+            li: List of strings/sentences.
+
+        Returns:
+            List of strings/sentences with hyphenated words joined into a single word.
+
+        """
+        current = None
+        new_list = copy.deepcopy(li)
+        for ix, l in enumerate(li):
+            regex_match = re.search(r"\w+(-|–){1}$", l)
+
+            if current:
+                word_fragment = current.rstrip("-")
+                # TODO: Handle non-English words
+                if word_fragment in english_words_set:
+                    # Check if the word with hyphen removed is an english
+                    # word and if it is, make this the first word of the newline
+                    # without the hyphen (e.g. repair-ing)
+                    newline_first_word = word_fragment + l.split(" ")[0].lstrip("-")
+                    if newline_first_word in english_words_set:
+                        new_list[ix] = newline_first_word + l.split(" ")[1:]
+                    # Otherwise, keep the hyphenation but put it on a newline e.g. post-processing.
+                    else:
+                        new_list[ix] = word_fragment + "-" + l
+                else:
+                    new_list[ix] = word_fragment + l
+                # Reset.
+                current = None
+            if regex_match:
+                # Strip matching regex from the end of the string.
+                new_list[ix] = l[: regex_match.start()]
+                current = regex_match[0]
+        return new_list
+
+    def process(self, contents: Document) -> Document:
+        contents = contents.to_dict()
+        for ix, page in enumerate(contents["pages"]):
+            for text_block in page["text_blocks"]:
+                text_block["text"] = self._rewrap_hyphenated_words(text_block["text"])
+        contents = Document.from_dict(contents)
+        return contents
+
+
+class AdobeTextStylingPostProcessor(PostProcessor):
+    """
+    Some semantic passages have separate blocks for styling markers such
+    as underlines, superscripts and subscripts. We want to group such cases
+    into single contiguous blocks. However, we want to keep the styling information
+    because it's often relevant to semantics. For example, CO2 is often represented
+    using a subscript. For now, we keep everything and indicate the styling with
```

Comment:
I think the 'indicating the styling with HTML tags inline' part of this docstring needs updating